xonsh/parsers/base.py

```diff
@@ -3085,7 +3149,19 @@ def p_subproc_arg_single(self, p):
     def p_subproc_arg_many(self, p):
         """subproc_arg : subproc_arg subproc_arg_part"""
         # This glues the string together after parsing
-        p[0] = p[1] + p[2]
+        p1 = p[1]
+        p2 = p[2]
+        if isinstance(p1, ast.Str) and isinstance(p2, ast.Str):
+            p0 = ast.Str(p1.s + p2.s, lineno=p1.lineno, col_offset=p1.col_offset)
+        elif isinstance(p1, list):
```

Comment:
Is there a case here where both `p1` and `p2` are `list`s?  Does that need to be cased?

Reply:
I think that isn't possible given how the parser reads tokens from left to right, but it is good to be robust here.  Fix incoming.

Reply:
Fix pushed up!